app/lib/actions/pulp3/repository/upload_file.rb

```diff
@@ -34,10 +35,21 @@ def invoke_external_task
 
             if response
               upload_href = response.pulp_href
-              output[:pulp_tasks] = [uploads_api.commit(upload_href, sha256: sha256)]
+              #Check for any duplicate artifacts created in parallel subtasks
+              duplicate_sha_artifact_list = ::Katello::Pulp3::Api::Core.new(smart_proxy).artifacts_api.list("sha256": sha256)
+              duplicate_sha_artifact_href = duplicate_sha_artifact_list&.results&.first&.pulp_href
```

Comment:
upload file gets run in parallel it looks like, i'm kinda worried this could open us up to a race condition, but maybe its not worth worrying about right now.  Or maybe we should just switch upload files to upload them in sequence instead of in parallel?

Reply:
and that will decrease performance for web-ui based uploads, but those have to be small anyways, so its maybe not a big deal?

Reply:
Ya..this additional check at the last minute before commit is to avoid a race consition I was running into..There is a possibility of running into race conditions with duplicate files in the same batch even with this. But I think this is such a specific use-case that we should be OK? Thoughts?

Reply:
after some discussion we decided to serialize webui uploads